src/Simplify_Max.cpp

```diff
@@ -113,6 +113,7 @@ Expr Simplify::visit(const Max *op, ExprInfo *bounds) {
              rewrite(max(min(max(x, y), z), y), max(min(x, z), y)) ||
              rewrite(max(min(max(y, x), z), y), max(y, min(x, z))) ||
              rewrite(max(max(x, c0), c1), max(x, fold(max(c0, c1)))) ||
+             rewrite(max(max(x, y) + c0, x), max(x, y + c0), c0 < 0) ||
```

Comment:
Looks correct to me. While you're here, please also add the case where c0 > 0: `max(x, y) + c0`, and versions with x, y swapped in the innermost max, and the equivalent rules to Simplify_Min.